### PR TITLE
bpf/conntrack: fix assumed ordering in NewStaleNATScanner

### PR DESCRIPTION
**Pull request by tomastigera**
Legs A-B do not need to have the same ordering in fwd key and the rev key.

Port is also significant as there may be services with multiple ports.

## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: bpf/conntrack/cleanup.go [hunk 1]
```diff
@@ -318,18 +322,29 @@ func NewStaleNATScanner(frontendHasBackend NATChecker) EntryScanner {
...
 				epIP = revB
-				epPort = revKey.PortB()
+				epPort = revBport
 				svcIP = kB
-				svcPort = k.PortB()
-			} else if kA.Equal(revB) {
+				svcPort = kBport
+			} else if kB.Equal(revA) && kBport == revAport {
+				epIP = revB
+				epPort = revAport
```
Comment (fasaxc) [inline bpf/conntrack/cleanup.go:332]: `revBport`?

File: bpf/conntrack/cleanup.go [hunk 2]
```diff
@@ -318,18 +322,29 @@ func NewStaleNATScanner(frontendHasBackend NATChecker) EntryScanner {
...
+			} else if kA.Equal(revB) && kAport == revBport {
 				epIP = revA
-				epPort = revKey.PortA()
+				epPort = revAport
 				svcIP = kA
-				svcPort = k.PortA()
+				svcPort = kAport
+			} else if kB.Equal(revB) && kBport == revBport {
+				epIP = revA
+				epPort = revBport
```
Comment (fasaxc) [inline bpf/conntrack/cleanup.go:342]: revAport to go with the `epIP`?

File: bpf/conntrack/cleanup.go [hunk 3]
```diff
@@ -318,18 +322,29 @@ func NewStaleNATScanner(frontendHasBackend NATChecker) EntryScanner {
...
+			} else if kB.Equal(revA) && kBport == revAport {
+				epIP = revB
+				epPort = revAport
+				svcIP = kA
+				svcPort = kAport
+			} else if kA.Equal(revB) && kAport == revBport {
 				epIP = revA
-				epPort = revKey.PortA()
+				epPort = revAport
 				svcIP = kA
```
Comment (fasaxc) [inline bpf/conntrack/cleanup.go:338]: think this should be `kB`  if `kA == revB` then `kA` is the source pod IP si it's not the service IP

File: bpf/conntrack/cleanup.go [hunk 4]
```diff
@@ -318,18 +322,29 @@ func NewStaleNATScanner(frontendHasBackend NATChecker) EntryScanner {
...
-				svcPort = k.PortA()
+				svcPort = kAport
+			} else if kB.Equal(revB) && kBport == revBport {
+				epIP = revA
+				epPort = revBport
+				svcIP = kB
```
Comment (fasaxc) [inline bpf/conntrack/cleanup.go:343]: Think this should be `kA` since `kB.Equal(revB)` means that `kB` is the source pod IP.

**Comment by tomastigera** [comment]
The test now checks whether what de selected as front/back ends is what we expect

**Pull request merged**